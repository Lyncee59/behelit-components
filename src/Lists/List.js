import React from 'react'
import PropTypes from 'prop-types'
import styled from 'styled-components'

const Wrapper = styled.ul`
  list-style-type: none;
  margin: 0;
  padding-left: 10px;
  padding-top: 10px;
  padding-bottom: 10px;
`

const List = ({ children, ...rest }) => <Wrapper {...rest}>{children}</Wrapper>

List.propTypes = {
  children: PropTypes.node
}

export default List
