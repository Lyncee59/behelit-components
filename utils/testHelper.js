import React from 'react'
import { mount } from 'enzyme'
import { ThemeProvider } from '../src/Theme'

export const mountWithTheme = (tree) => mount(<ThemeProvider>{tree}</ThemeProvider>)