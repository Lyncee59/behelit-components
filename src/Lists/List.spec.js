import React from 'react'
import { mountWithTheme } from '../../utils/testHelper'

import List from './List'

describe('List', () => {
  it('should render correctly', () => {
    const tree = mountWithTheme(<List>List</List>)

    expect(tree.find(List)).toMatchSnapshot()
  })
})
