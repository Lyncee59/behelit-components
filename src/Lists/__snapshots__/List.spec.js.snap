// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`List should render correctly 1`] = `
.c0 {
  list-style-type: none;
  margin: 0;
  padding-left: 10px;
  padding-top: 10px;
  padding-bottom: 10px;
}

<List>
  <styled.ul>
    <StyledComponent
      forwardedComponent={
        Object {
          "$$typeof": Symbol(react.forward_ref),
          "attrs": Array [],
          "componentStyle": ComponentStyle {
            "componentId": "sc-bdVaJa",
            "isStatic": true,
            "lastClassName": "c0",
            "rules": Array [
              "
  list-style-type: none;
  margin: 0;
  padding-left: 10px;
  padding-top: 10px;
  padding-bottom: 10px;
",
            ],
          },
          "displayName": "styled.ul",
          "foldedComponentIds": Array [],
          "render": [Function],
          "styledComponentId": "sc-bdVaJa",
          "target": "ul",
          "toString": [Function],
          "warnTooManyClasses": [Function],
          "withComponent": [Function],
        }
      }
      forwardedRef={null}
    >
      <ul
        className="c0"
      >
        List
      </ul>
    </StyledComponent>
  </styled.ul>
</List>
`;
